be/src/runtime/stream_load/transaction_mgr.h

```diff
@@ -0,0 +1,109 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+
+#include "stream_load_context.h"
+#include "util/starrocks_metrics.h"
+#include "util/thread.h"
+#include "util/uid_util.h" // for std::hash for UniqueId
+
+namespace starrocks {
+
+class HttpRequest;
+
+// used to register all streams in process so that other module can get this stream
+class StreamContextMgr {
+public:
+    StreamContextMgr() {}
+
+    ~StreamContextMgr() {
+        std::lock_guard<std::mutex> l(_lock);
+        for (auto& iter : _stream_map) {
+            if (iter.second->unref()) {
+                delete iter.second;
+            }
+        }
+    }
+
+    Status put(const std::string& id, StreamLoadContext* stream) {
+        std::lock_guard<std::mutex> l(_lock);
+        auto it = _stream_map.find(id);
+        if (it != std::end(_stream_map)) {
+            return Status::InternalError("id already exist");
+        }
+        _stream_map.emplace(id, stream);
+        stream->ref();
+        VLOG(3) << "put stream load pipe: " << id;
+        return Status::OK();
+    }
+
+    StreamLoadContext* get(const std::string& id) {
+        std::lock_guard<std::mutex> l(_lock);
+        auto it = _stream_map.find(id);
+        if (it == std::end(_stream_map)) {
+            return nullptr;
+        }
+        auto stream = it->second;
+        stream->ref();
+        return stream;
+    }
+
+    void remove(const std::string& id) {
```

Comment:
If the transaction timeout, you should call remove().
Otherwise, the _stream_map will expand

Reply:
there is a background thread to do this